src/lpm.c

```diff
@@ -95,6 +95,9 @@ lpm_clear(lpm_t *lpm, lpm_dtor_t dtor, void *arg)
 		hmap->nitems = 0;
 	}
 	memset(lpm->bitmask, 0, sizeof(lpm->bitmask));
+	if (dtor) {
+		dtor(arg, NULL, 0, lpm->defval);
```

Comment:
Just to keep the API promise:
```
static const uint32_t zero_address[LPM_MAX_WORDS];
dtor(arg, &zero_address, 0, lpm->defval);
```

Reply:
Seems reasonable, I'll change that.

However, while we're here, there is another problem with the way `defval` is used - unlike all other prefix lengths, the zero length prefix doesn't differentiate between _addresses of different lengths_. i.e. `0.0.0.0/0` and `::/0` resolve to the same value, however `127.0.0.0/8` and `7f00::/8` don't.

So I'm wondering if defval should be changed to an array of void * indexed by address length, or something along those lines.

Reply:
Yes, I would say:
```
struct lpm {
	...
	void *		defval[2];
	...
};

#define	IPVER_IDX(len)	((len) >> 4)

lpm->defval[IPVER_IDX(len)] = val;
```
We can sprinkle some `ASSERT(len == 4 || len == 16);`, since library supports only IPv4 or IPv6 addresses (Lua code checks for these lengths already).

Reply:
Interestingly, as far as I can tell, `lpm.c` doesn't care what the length of the address is, so long as it's less than 16 bytes, so in theory you could also shove MAC addresses or FireWire hardware addresses in there too.

Not that it serves much of a purpose, mind you, unless you wanted to lookup what NIC manufacturer some frames were created on or something.

Reply:
Yes, I kept the code agnostic just because it can be that way. However, there is no point to add complexity just to keep it agnostic. I do not think anybody is interested in LPM for MAC addresses. ;)

Reply:
Arguably, it'd be simpler to just use length as the index -- preserving the agnosticism at the cost of `14 * sizeof(void *)` bytes more.

    #define	LPM_MAX_BYTES      ((LPM_MAX_PREFIX) >> 3)

    struct lpm {
	    ...
	    void *		defval[LPM_MAX_BYTES];
	    ...
    };

    lpm->defval[len] = val;

Reply:
Yes, but it would also span over more cache lines. Space is not a concern, but dense data structures result in fewer cache misses and tend to be faster.

Reply:
Fair enough, 2-element array it is, then. Updating...